### PR TITLE
feat: Support v2 protos [DEV-1912]

### PR DESCRIPTION
**Pull request by DaevMithran**

### CODE REVIEW

File: types/did_doc.go [hunk 1]
```diff
@@ -19,22 +19,21 @@ type DidDoc struct {
...
-	Id                 string            `json:"id,omitempty"`
-	Type               string            `json:"type,omitempty"`
-	Controller         string            `json:"controller,omitempty"`
-	PublicKeyJwk       map[string]string `json:"publicKeyJwk,omitempty"`
-	PublicKeyMultibase string            `json:"publicKeyMultibase,omitempty"`
+	Context              []string `json:"@context,omitempty"`
+	Id                   string   `json:"id,omitempty"`
+	Type                 string   `json:"type,omitempty"`
+	Controller           string   `json:"controller,omitempty"`
+	VerificationMaterial string   `json:"verificationMaterial,omitempty"`
```
Comment (askolesov) [inline types/did_doc.go:26]: Here we need additional logic. `VerificationMaterial` contains json. All fields from this json should be added to the verification method.

File: types/did_doc_metadata_test.go [hunk 2]
```diff
@@ -26,32 +26,31 @@ func TestNewResolutionDidDocMetadata(t *testing.T) {
...
-		Created:           resourceHeader.Created,
-		Checksum:          FixResourceChecksum(resourceHeader.Checksum),
+		ResourceURI:       validDid + RESOURCE_PATH + resourceMetadata.Id,
+		CollectionId:      resourceMetadata.CollectionId,
+		ResourceId:        resourceMetadata.Id,
+		Name:              resourceMetadata.Name,
+		ResourceType:      resourceMetadata.ResourceType,
+		MediaType:         resourceMetadata.MediaType,
+		Created:           resourceMetadata.Created,
+		Checksum:          FixResourceChecksum(resourceMetadata.Checksum),
```
Comment (askolesov) [inline types/did_doc_metadata_test.go:36]: No need for the fix anymore.

File: go.mod [hunk 3]
```diff
@@ -3,106 +3,110 @@ module github.com/cheqd/did-resolver
 go 1.17
 
 require (
-	github.com/cheqd/cheqd-node v0.6.0
+	github.com/cheqd/cheqd-node v1.0.0-develop.10
```
Comment (askolesov) [inline go.mod:6]: Reminder - change to v1.0.0 once released.

**Review by askolesov — COMMENTED** [review]
Looks really good!

File: types/did_doc_metadata_test.go (hunk 2 above)
Comment (DaevMithran) [reply types/did_doc_metadata_test.go:36]: The checkSum is in bytes, the Fix just converts it to a string for the resolved output.

File: types/did_doc.go (hunk 1 above)
Comment (DaevMithran) [reply types/did_doc.go:26]: @askolesov Do you mean the resolved result should have the fields inside parsed verificationMaterial, This struct will remain the same right?

**Comment by DaevMithran** [comment]
@askolesov can you check why the golang linter is failing, looks like some dependency issue. Thanks

**Comment by ankurdotb** [comment]
Task linked: [DEV-1912 Update existing did-resolver to use latest version of cheqd-node / latest protobuf definitions](https://app.clickup.com/t/6600954/DEV-1912)

File: types/did_doc.go (hunk 1 above)
Comment (askolesov) [reply types/did_doc.go:26]: Yeah... Look, if we have this stored on ledger:
```
{
Id
Type
...
VerificationMaterial: "{fieldXXX: valueX, fieldYYY: valueY}"
}
```
We want to get this on the resolver:
```
{
Id
Type
...
fieldXXX: valueX
fieldYYY: valueY
}
```

Comment (DaevMithran) [reply types/did_doc.go:26]: Do you have any ideas on implementing this ?, Its not possible to have an anonymous field with interface{} type.

File: go.mod (hunk 3 above)
Comment (ankurdotb) [reply go.mod:6]: Change it to more recent dev release

**Review by ankurdotb — CHANGES_REQUESTED** [review]
See ClickUp for changes requested/to consider

File: types/did_doc.go (hunk 1 above)
Comment (DaevMithran) [reply types/did_doc.go]: This is complete

**Comment by Tweeddalex** [comment]
@DaevMithran is there anything blocking this being merged?

**Comment by abdulla-ashurov** [comment]
> @DaevMithran is there anything blocking this being merged?

I implemented all the API except get AllDidDocVersionsMetadata

**Comment by abdulla-ashurov** [comment]
> @DaevMithran is there anything blocking this being merged?

Done! ✅

**Pull request merged**